src/database/PostgreSQL.ts

```
@@ -161,22 +161,36 @@ export class PostgreSQL implements IDatabase {
     });
   }
 
-  cleanSaves(game_id: GameId, save_id: number): void {
-    // DELETE all saves except initial and last one
-    this.client.query('DELETE FROM games WHERE game_id = $1 AND save_id < $2 AND save_id > 0', [game_id, save_id], (err) => {
+  getMaxSaveId(game_id: GameId, cb: DbLoadCallback<number>): void {
+    this.client.query('SELECT MAX(save_id) FROM games WHERE game_id = $1', [game_id], (err: Error | null, res: QueryResult<any>) => {
       if (err) {
-        console.error('PostgreSQL:cleanSaves', err);
-        throw err;
+        return cb(err ?? undefined, undefined);
       }
-      // Flag game as finished
-      this.client.query('UPDATE games SET status = \'finished\' WHERE game_id = $1', [game_id], (err2) => {
-        if (err2) {
-          console.error('PostgreSQL:cleanSaves2', err2);
-          throw err2;
-        }
-      });
+      cb(undefined, res.rows[0]);
     });
-    this.purgeUnfinishedGames();
+  }
+
+  throwIf(err: any, condition: string) {
+    if (err) {
+      console.error('PostgreSQL', condition, err);
+      throw err;
+    }
+  }
+
+  cleanSaves(game_id: GameId): void {
+    this.getMaxSaveId(game_id, ((err, save_id) => {
+      this.throwIf(err, 'cleanSaves0');
```

With how this is written `save_id` could also be `undefined`. I would also throw an error if `save_id` were `undefined`.

---

> With how this is written `save_id` could also be `undefined`. I would also throw an error if `save_id` were `undefined`.

OK. Doing that now.